modules/failure-handler/src/test/java/org/apache/ignite/internal/failure/FailureProcessorTest.java

```diff
@@ -49,4 +53,21 @@ void testFailureProcessing() {
             failureProcessor.stop();
         }
     }
+
+    @Test
+    void testIgnoredFailureTypes() {
+        FailureHandler handler = new NoOpFailureHandler();
+
+        FailureProcessor failureProcessor = new FailureProcessor("node_name", handler);
+
+        assertThat(failureProcessor.start(), willSucceedFast());
+
+        try {
+            assertThat(failureProcessor.process(new FailureContext(SYSTEM_WORKER_BLOCKED, null)), is(false));
+
+            assertThat(failureProcessor.process(new FailureContext(SYSTEM_CRITICAL_OPERATION_TIMEOUT, null)), is(false));
+        } finally {
+            failureProcessor.stop();
+        }
```

Comment:
```suggestion
        try {
            assertThat(failureProcessor.start(), willSucceedFast());
        
            assertThat(failureProcessor.process(new FailureContext(SYSTEM_WORKER_BLOCKED, null)), is(false));

            assertThat(failureProcessor.process(new FailureContext(SYSTEM_CRITICAL_OPERATION_TIMEOUT, null)), is(false));
        } finally {
            failureProcessor.stop();
        }
```

Reply:
Honestly, it is not obvious from the javadoc that it is allowed to call `stop` on non started component. Anyway. I fixed it in the way you proposed.